### PR TITLE
Read reference by id (API)

### PR DESCRIPTION
**Pull request by mrkvon**
`GET /api/references/:referenceId`

Continued work on #718 
Depends on #759 and #723

### CODE REVIEW

File: modules/references/server/controllers/reference.server.controller.js [hunk 1]
```diff
@@ -326,3 +326,65 @@ exports.readMany = function readMany(req, res, next) {
...
+    // validate
+    validate.bind(this, validateReadOne, id),
+    // find the reference by id
+    function (cb) {
+      Reference.findById(req.params.referenceId)
+        .select(referenceFields)
+        .populate('userFrom userTo', userProfile.userMiniProfileFields)
+        .exec(cb);
+    },
+    function (reference, cb) {
```
Comment (mrkvon) [inline modules/references/server/controllers/reference.server.controller.js]:
```suggestion

    // User can read public references only. Or references that she created.
    function (reference, cb) {
```

File: modules/references/server/controllers/reference.server.controller.js [hunk 2]
```diff
@@ -345,3 +350,74 @@ exports.readMany = function readMany(req, res, next) {
...
+            details: {
+              reference: 'not found'
+            }
+          }
+        });
+      }
+
+      // assign the reference to the request object
+      // when reference is public, only userFrom can see it whole
+      var isUserFrom = reference.userFrom._id.toString() === req.user._id.toString();
```
Comment (simison) [inline modules/references/server/controllers/reference.server.controller.js]: I believe you can compare Mongo ObjectIds with `userFrom._id.equals(req.user._id)`

File: modules/references/tests/server/reference-read-one.server.routes.tests.js [hunk 3]
```diff
@@ -1,19 +1,234 @@
...
+      users = usrs;
+      return done(err);
+    });
+  });
+
+  /**
+   *   0 1 2
+   * 0 . T F
+   * 1 F . T
+   * 2 T F .
```
Comment (simison) [inline modules/references/tests/server/reference-read-one.server.routes.tests.js]: I get this is ids and such but the comment could be a little bit less cryptic. :-)

File: modules/references/server/controllers/reference.server.controller.js (hunk 1 above)
Comment (mrkvon) [reply modules/references/server/controllers/reference.server.controller.js]: different comment already added

File: modules/references/server/controllers/reference.server.controller.js (hunk 2 above)
Comment (mrkvon) [reply modules/references/server/controllers/reference.server.controller.js]: This and other similar comparisons fixed in a4f41ec.

File: modules/references/tests/server/reference-read-one.server.routes.tests.js (hunk 3 above)
Comment (mrkvon) [reply modules/references/tests/server/reference-read-one.server.routes.tests.js]: Improved in a4f41ec

**Comment by simison** [comment]
Noted missing email templates here: https://github.com/Trustroots/trustroots/issues/849

**Pull request merged**